parachain/runtime/testnet-interlay/src/lib.rs

```diff
@@ -1419,39 +1431,20 @@ impl_runtime_apis! {
         fn dispatch_benchmark(
             config: frame_benchmarking::BenchmarkConfig
         ) -> Result<Vec<frame_benchmarking::BenchmarkBatch>, sp_runtime::RuntimeString> {
-            use frame_benchmarking::{Benchmarking, BenchmarkBatch, add_benchmark, TrackedStorageKey};
-
-            let whitelist: Vec<TrackedStorageKey> = vec![
-                // Block Number
-                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac").to_vec().into(),
-                // Total Issuance
-                hex_literal::hex!("c2261276cc9d1f8598ea4b6a74b15c2f57c875e4cff74148e4628f264b974c80").to_vec().into(),
-                // Execution Phase
-                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a").to_vec().into(),
-                // Event Count
-                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850").to_vec().into(),
-                // System Events
-                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7").to_vec().into(),
-            ];
+            use frame_benchmarking::{Benchmarking, BenchmarkBatch, TrackedStorageKey};
+
+            use frame_support::traits::WhitelistedStorageKeys;
+            let mut whitelist: Vec<TrackedStorageKey> = AllPalletsWithSystem::whitelisted_storage_keys();
+
+            // Treasury Account
+            // TODO: this is manual for now, someday we might be able to use a
+            // macro for this particular key
+            let treasury_key = frame_system::Account::<Runtime>::hashed_key_for(TreasuryAccount::get());
+            whitelist.push(treasury_key.to_vec().into());
 
             let mut batches = Vec::<BenchmarkBatch>::new();
             let params = (&config, &whitelist);
-
-            add_benchmark!(params, batches, btc_relay, BTCRelay);
-            add_benchmark!(params, batches, fee, Fee);
-            add_benchmark!(params, batches, farming, Farming);
-            add_benchmark!(params, batches, issue, Issue);
-            add_benchmark!(params, batches, loans, Loans);
-            add_benchmark!(params, batches, nomination, Nomination);
-            add_benchmark!(params, batches, oracle, Oracle);
-            add_benchmark!(params, batches, redeem, Redeem);
-            add_benchmark!(params, batches, replace, Replace);
-            add_benchmark!(params, batches, vault_registry, VaultRegistry);
-            add_benchmark!(params, batches, dex_general, DexGeneral);
-            add_benchmark!(params, batches, dex_stable, DexStable);
-            add_benchmark!(params, batches, dex_swap_router, DexSwapRouter);
-
-            if batches.is_empty() { return Err("Benchmark not found for this pallet.".into()) }
+            add_benchmarks!(params, batches);
             Ok(batches)
         }
```

Comment:
Shouldn't we also add unit tests to check the whitelist in the testnet runtimes?

Reply:
Not worth it, we will remove those runtimes soon